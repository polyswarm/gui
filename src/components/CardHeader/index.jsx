// Vendor Imports
import React, {Component} from 'react';
import PropTypes from 'prop-types';
// Bounty Management Imports

// Component Imports

class CardHeader extends Component {
  render() {
    const {props: {title, subhead}} = this;
    return (
      <header className='CardHeader'>
        {title}
        {subhead && subhead.length > 0 && (
          <p className='CardSubHeader'>
            {subhead}
          </p>
        )}
      </header>
    );
  }
}
CardHeader.proptypes = {
  title: PropTypes.string,
  subhead: PropTypes.string,
}
export default CardHeader;