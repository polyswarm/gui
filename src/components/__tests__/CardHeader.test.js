import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import CardHeader from '../CardHeader';

it('renders without crashing', () => {
  const wrapper = render(<CardHeader />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('shows the title when passed', () => {
  const wrapper = mount(
    <CardHeader title='asdf'/>
  );

  expect(wrapper.text()).toEqual('asdf');
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

