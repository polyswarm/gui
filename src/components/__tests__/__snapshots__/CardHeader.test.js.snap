// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<header
  class="CardHeader"
>
  
</header>
`;

exports[`shows the title when passed 1`] = `null`;
